### PR TITLE
Add saws python package

### PR DESCRIPTION
**Pull request by martinklepsch**
Hello. This is my first real PR to nixpkgs.

It's not 100% done yet I'm afraid but now is probably a good time to get feedback/advice.

The last hurdle I'm facing is that when trying to install `python3.4-saws-0.1.1` it fails because it cant find `awscli`. For some reason this works fine with `python2.7-saws-0.1.1`. I'm lost on the quest figuring out why that is.

```
Searching for awscli>=1.7.46

Note: Bypassing https://pypi.python.org/simple/awscli/ (disallowed host; see http://bit.ly/1dg9ijs for details).

Couldn't find index page for 'awscli' (maybe misspelled?)
Scanning index of all packages (this may take a while)

Note: Bypassing https://pypi.python.org/simple/ (disallowed host; see http://bit.ly/1dg9ijs for details).

No local packages or download links found for awscli>=1.7.46
```

We should probably wait merging this until https://github.com/donnemartin/saws/issues/22 is done.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -18136,6 +18191,32 @@ let
...
   };
 
+  saws = buildPythonPackage rec {
+    name = "saws-0.1.1";
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/s/saws/saws-0.1.1.tar.gz";
+      md5 = "e81437360466216a41db6c78d9b46aae";
+    };
+
+    propagatedBuildInputs = with self; [ awscli pygments six prompt_toolkit fuzzyfinder configobj click ]
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: use `pkgs.awscli` to avoid extra argument to this file

**Comment by martinklepsch** [comment]
@domenkozar even with the change you suggested I keep getting
Even with this change I keep getting:
```
Searching for awscli>=1.7.46

Note: Bypassing https://pypi.python.org/simple/awscli/ (disallowed host; see http://bit.ly/1dg9ijs for details).

Couldn't find index page for 'awscli' (maybe misspelled?)
Scanning index of all packages (this may take a while)

Note: Bypassing https://pypi.python.org/simple/ (disallowed host; see http://bit.ly/1dg9ijs for details).

No local packages or download links found for awscli>=1.7.46
error: Could not find suitable distribution for Requirement.parse('awscli>=1.7.46')
```

**Comment by martinklepsch** [comment]
I'm also wondering: shouldn't `awscli` be packaged in `python-packages.nix`?

**Comment by domenkozar** [comment]
`awscli` is packaged with py27, so that's why it doesn't find the dependency.

**Comment by martinklepsch** [comment]
@domenkozar aws-cli is supposed to work with lots of python versions (https://github.com/aws/aws-cli#aws-cli) — why is it packaged with py27? Is that something I could have seen in the nix file?

**Comment by martinklepsch** [comment]
Hey @pesterhazy, maybe you can help me out a bit here? :smile:

**Comment by lethalman** [comment]
I think you can disable the build of saws for python3 for now, and let's go on.

**Comment by domenkozar** [comment]
yeah we need to move `pkgs/tools/admin/awscli/default.nix:` into `pkgs/top-level/python-packages.nix`

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -504,6 +504,22 @@ let
     };
   };
 
+  appnope = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: appnope is included in nixpkgs

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -3580,6 +3596,22 @@ let
     };
   };
 
+  fuzzyfinder = buildPythonPackage rec {
+    name = "fuzzyfinder-1.0.0";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/f/fuzzyfinder/fuzzyfinder-1.0.0.tar.gz";
+      md5 = "c76fac258987760c9e59d077bc699a06";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you determine a sha256 here?

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -7282,6 +7314,26 @@ let
...
+    };
+
+    propagatedBuildInputs = with self; [];
+
+    meta = {
+      description = "Stand-alone GNU readline module for Python ";
+      homepage = https://github.com/ludwigschwardt/python-gnureadline;
+      license = licenses.bsd3;
+      maintainers = with maintainers; [ fridh ];
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: I don't think I am the maintainer here? You could add yourself here if you would like to maintain this :-)

**Comment by domenkozar** [comment]
@martinklepsch ping :)

**Comment by martinklepsch** [comment]
Hey @domenkozar sorry for the delay on this. After some more playing I figured out that I actually don't have much use for saws. Also my nix install is somewhat broken since some recent update so I can't really do much right now.

Feel free to close this or fork & make it work :)

**Comment by domenkozar** [comment]
@martinklepsch what's broken exactly :)

**Comment by domenkozar** [comment]
Hmm I really don't want to close this as it's mostly done. I'll leave it open for now and try to get to it later.

**Comment by martinklepsch** [comment]
@domenkozar sorry for the unresponsiveness on my side! 

I ran into issues updating from 1.9 to 1.10 that I haven't had time to fix yet. I'm also on Darwin which makes this more challenging (at least for me right now). I hope to get back to using Nix more but right now I don't have the time :/

**Comment by domenkozar** [comment]
@martinklepsch darwin on master should be in a pretty good state now. Do open an issue if something blocks you :)

**Comment by FRidh** [comment]
Please open again when you find the time to finish this :-)

**Pull request closed without merging**